diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100644
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,10 @@
+module(
+    name = "xnnpack",
+    version = "0.0.0-20220831-e8f74a9",
+)
+
+bazel_dep(name = "bazel_skylib", version = "1.7.1")
+bazel_dep(name = "cpuinfo", version = "0.0.0-20220819-8ec7bd9")
+bazel_dep(name = "fp16", version = "0.0.0-20210320-0a92994", repo_name = "FP16")
+bazel_dep(name = "fxdiv", version = "0.0.0-20201209-63058ef", repo_name = "FXdiv")
+bazel_dep(name = "pthreadpool", version = "0.0.0-20230829-4fe0e1e")
diff --git i/build_defs.bzl w/build_defs.bzl
index b8217a18d..7432a237f 100644
--- i/build_defs.bzl
+++ w/build_defs.bzl
@@ -202,6 +202,7 @@ def xnnpack_cc_library(
             ":linux_armv7a": ["-lpthread"],
             ":linux_arm64": ["-lpthread"],
             ":android": ["-lm"],
+            ":android_x86_64": ["-lm"],
             "//conditions:default": [],
         }),
         textual_hdrs = hdrs,
diff --git a/build_defs.bzl b/build_defs.bzl
index b8217a1..3e468fe 100755
--- a/build_defs.bzl
+++ b/build_defs.bzl
@@ -20,7 +20,7 @@ def xnnpack_gcc_std_copts():
 
 def xnnpack_msvc_std_copts():
     """MSVC compiler flags to specify language standard for C sources."""
-    return ["/Drestrict="]
+    return ["-Drestrict="]
 
 def xnnpack_std_cxxopts():
     """Compiler flags to specify language standard for C++ sources."""
@@ -163,7 +163,7 @@ def xnnpack_cc_library(
             ":windows_x86_64_clang": ["/clang:" + opt for opt in gcc_x86_copts],
             ":windows_x86_64_mingw": mingw_copts + gcc_x86_copts,
             ":windows_x86_64_msys": msys_copts + gcc_x86_copts,
-            ":windows_x86_64": msvc_x86_64_copts,
+            ":windows_x86_64": msvc_x86_64_copts + gcc_x86_copts,
             ":android_armv7": aarch32_copts,
             ":android_arm64": aarch64_copts,
             ":android_x86": gcc_x86_copts,
diff --git i/build_defs.bzl w/build_defs.bzl
index b8217a18d..b17b23649 100644
--- i/build_defs.bzl
+++ w/build_defs.bzl
@@ -152,7 +152,6 @@ def xnnpack_cc_library(
             "-Iinclude",
             "-Isrc",
         ] + copts + select({
-            ":linux_k8": gcc_x86_copts,
             ":linux_arm": aarch32_copts,
             ":linux_armeabi": aarch32_copts,
             ":linux_armhf": aarch32_copts,
@@ -195,7 +194,6 @@ def xnnpack_cc_library(
         includes = ["include", "src"] + includes,
         linkstatic = True,
         linkopts = select({
-            ":linux_k8": ["-lpthread"],
             ":linux_arm": ["-lpthread"],
             ":linux_armeabi": ["-lpthread"],
             ":linux_armhf": ["-lpthread"],
diff --git a/BUILD.bazel b/BUILD.bazel
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -16268,33 +16268,34 @@ config_setting(
     },
 )
 
-config_setting(
+selects.config_setting_group(
     name = "emscripten",
-    values = {"crosstool_top": "@emsdk//emscripten_toolchain:everything"},
+    match_any = [
+        ":emscripten_wasm",
+        ":emscripten_wasmsimd",
+        ":emscripten_wasmrelaxedsimd",
+    ],
 )
 
 config_setting(
     name = "emscripten_wasm",
     values = {
-        "crosstool_top": "@emsdk//emscripten_toolchain:everything",
-        "cpu": "wasm",
+        "cpu": "wasm32",
     },
 )
 
 config_setting(
     name = "emscripten_wasmsimd",
     values = {
-        "crosstool_top": "@emsdk//emscripten_toolchain:everything",
-        "cpu": "wasm",
-        "features": "wasm_simd",
+        "cpu": "wasm32",
+        "features": "simd",
     },
 )
 
 config_setting(
     name = "emscripten_wasmrelaxedsimd",
     values = {
-        "crosstool_top": "@emsdk//emscripten_toolchain:everything",
-        "cpu": "wasm",
-        "features": "wasm_relaxed_simd",
+        "cpu": "wasm32",
+        "features": "relaxed_simd",
     },
 )
